torch/autograd/graph.py

```diff
@@ -5,15 +5,121 @@
 from torch.utils._python_dispatch import TorchDispatchMode
 from collections import defaultdict
 import weakref
+import abc
 
 __all__ = [
     "saved_tensors_hooks",
     "save_on_cpu",
     "disable_saved_tensors_hooks",
     "register_multi_grad_hook",
     "allow_mutation_on_saved_tensors",
+    "Node",
 ]
 
+class Node(abc.ABC):
+    @abc.abstractmethod
+    def name(self) -> str:
+        r"""Returns the name.
+
+        Example::
+            >>> import torch
+            >>> a = torch.tensor([0., 0., 0.], requires_grad=True)
+            >>> b = a.clone()
+            >>> assert isinstance(b.grad_fn, torch.autograd.graph.Node)
+            >>> print(b.grad_fn.name())
+            CloneBackward0
+        """
+        ...
+
+    @property
+    @abc.abstractmethod
+    def next_functions(self) -> Tuple[Tuple[Any, int], ...]:
```

Comment:
The first element in the inner tuple is an optional Node no?

Reply:
Yes, but since this method is inside the definition of the class, the class has not been defined yet, so I'm unable to refer to it there. (Maybe there is a better way?)

Reply:
You can use string to do that in python. For example in nn.Module we have `def add_module(self, name: str, module: Optional['Module']) -> None:`

Reply:
very cool, TIL!